config/config_test.go

```diff
@@ -69,12 +100,88 @@ func TestUpdate_Validate(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "No Duplicate Test Groups",
+			input: configpb.Configuration{
+				Dashboards: []*configpb.Dashboard{
+					{
+						Name: "dashboard_1",
+					},
+				},
+				TestGroups: []*configpb.TestGroup{
+					{
+						Name: "test_group_1",
+					},
+					{
+						Name: "TEST_GROUP_1",
+					},
+				},
+			},
+			expectedDup: map[string]bool{"testgroup1": true},
+		},
+		{
+			name: "No Duplicate Tabs in Dashboard",
+			input: configpb.Configuration{
+				Dashboards: []*configpb.Dashboard{
+					{
+						Name: "dashboard_1",
+						DashboardTab: []*configpb.DashboardTab{
+							{
+								Name: "tab_1",
+							},
+							{
+								Name: "TAB_1",
+							},
+							{
+								Name: "tab_2",
+							},
+						},
+					},
+					{
+						Name: "dashboard_2",
+						// Same tab name in a different dashboard is fine.
+						DashboardTab: []*configpb.DashboardTab{
+							{
+								Name: "tab_2",
+							},
+						},
+					},
+				},
+				TestGroups: []*configpb.TestGroup{
+					{
+						Name: "test_group_1",
+					},
+				},
+			},
+			expectedDup: map[string]bool{"tab1": true},
+		},
+		{
+			name: "No Duplicate Dashboard or Dashboard Group",
+			input: configpb.Configuration{
+				Dashboards: []*configpb.Dashboard{
+					{
+						Name: "name_1",
+					},
+				},
+				DashboardGroups: []*configpb.DashboardGroup{
+					{
+						Name: "name_1",
+					},
+				},
+				TestGroups: []*configpb.TestGroup{
+					{
+						Name: "test_group_1",
+					},
+				},
+			},
+			expectedDup: map[string]bool{"name1": true},
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			err := Validate(test.input)
-			if err != nil && test.expectedMissing == "" {
+			if err != nil && test.expectedMissing == "" && len(test.expectedDup) <= 0 {
```

Comment:
Prefer `len(test.expectedDup) == 0`. I'm pretty sure there's no situation in which len return a negative number, and I've already flipped the < twice in my head.